src/storage/tile_group.cpp

```diff
@@ -314,6 +315,37 @@ oid_t TileGroup::InsertTupleFromCheckpoint(oid_t tuple_slot_id,
   return tuple_slot_id;
 }
 
+// Compress all the tiles in the TileGroup only if the TileGroup is full.
+// If any tile is successfully compressed, set the compression status to true.
+// A Compressed tile is immutable.
+void TileGroup::CompressTiles() {
+  oid_t num_tiles = tiles.size();
+  oid_t next_tuple_slot = tile_group_header->GetCurrentNextTupleSlot();
+
+  if (next_tuple_slot < num_tuple_slots) {
+    LOG_TRACE("TileGroup is currently not full. Skipping compression.");
+    return;
+  }
+
+  for (oid_t i = 0; i < num_tiles; i++) {
+    LOG_TRACE("Compressing Tile %d in Tile Group", i);
+    Tile *old_tile = tiles[i].get();
+    std::shared_ptr<CompressedTile> new_tile(new CompressedTile(
+        backend_type, old_tile->GetHeader(), *(old_tile->GetSchema()),
+        old_tile->GetTileGroup(), old_tile->GetAllocatedTupleCount()));
+
+    new_tile->CompressTile(old_tile);
+
+    if (new_tile->IsCompressed()) {
+      tiles[i].reset();
+      tiles[i] = new_tile;
```

Comment:
There are multiple ways to release a shared_ptr. It can be done through going out of scope, assigning another shared_ptr or calling `reset`. Here I think calling `reset` is unnecessary in both if and else branch.